docs/erc-7824.md

```diff
@@ -132,14 +126,15 @@ For signature verification, the state hash is computed as:
 bytes32 stateHash = keccak256(
     abi.encode(
         channelId,
-        state.data,
+        state.intent,
         state.version,
+        state.data,
         state.allocations
     )
 );
 ```
 
-Note: The stateHash is bare signed without EIP-191 since the protocol is intended to be chain-agnostic.
+Note: The smart contract supports all popular signature formats, specifically: raw ECDSA, EIP-191, EIP-712, EIP-1271, and EIP-6492.
```

Comment:
Maybe, add the page with all of the supported signatures and message formats.

Because if I'm understanding correctly format is very important for some of them (EIP-712)